src/pip/_internal/metadata/pkg_resources.py

```diff
@@ -16,6 +17,7 @@ class Distribution(BaseDistribution):
     def __init__(self, dist):
         # type: (pkg_resources.Distribution) -> None
         self._dist = dist
+        self._version = None
```

Comment:
Hmm, I’m surprised Mypy is actually fine with this. I thought we need to mark the type explicitly here.

Reply:
Ah, the cause is #9502. Changing `[mypy-pip/_vendor/*] follow_imports = silent` correctly makes Mypy report:

```
src/pip/_internal/metadata/pkg_resources.py:50: error: Incompatible types in
assignment (expression has type "Union[LegacyVersion, Version]", variable has
type "None")
                self._version = parse_version(self._dist.version)
                                ^
src/pip/_internal/metadata/pkg_resources.py:51: error: Incompatible return
value type (got "None", expected "_BaseVersion")
            return self._version
```

So I guess there’s not really much we can do here.

Reply:
@stefanor I would prefer dropping the `self._version` cache from these places. `parse_version` is pretty fast and neglectable in most places, and pip doesn’t really access that attribute in too many places anyway.